src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -581,8 +592,17 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
             //if the id is not a guid, makes a guid based on the id of the model
             Guid connectorId = GuidUtility.tryParseOrCreateGuid(obj["Id"].Value<string>());
-
-            return new ConnectorModel(startPort, endPort, connectorId);
+            if(startPort != null && endPort != null)
+            {
+                return new ConnectorModel(startPort, endPort, connectorId);
+            }
+            else
+            {   //TODO should we log a notification to alert user that graph is missing nodes?
```

Comment:
I believe what you're logging underneath is accurate enough, even though it indicates node missing 99%. But we should stick to it I think